src/main/java/com/cst438/service/RegistrarServiceProxy.java

```diff
@@ -0,0 +1,175 @@
+                        s.setSecId(dto.secId());
+                        s.setBuilding(dto.building());
+                        s.setRoom(dto.room());
+                        s.setTimes(dto.times());
+                        s.setInstructor_email(dto.instructorEmail());
+                        sectionRepository.save(s);
+                    }
+                }
+
+            } else if (key.equals("addUser")) {
+                UserDTO dto = fromJsonString(parts[1], UserDTO.class);
+                User u = new User();
+                u.setId(dto.id());
+                u.setName(dto.name());
+                u.setEmail(dto.email());
+                u.setType(dto.type());
+                userRepository.save(u);
+
+            } else if (key.equals("deleteUser")) {
+                userRepository.deleteById(Integer.parseInt(parts[1]));
+
+            } else if (key.equals("updateUser")) {
+                UserDTO dto = fromJsonString(parts[1], UserDTO.class);
+                User u = userRepository.findById(dto.id()).orElse(null);
+                if (u != null) {
+                    u.setName(dto.name());
+                    u.setEmail(dto.email());
+                    u.setType(dto.type());
+                    userRepository.save(u);
+                }
+
+            } else if (key.equals("addEnrollment")) {
+                EnrollmentDTO dto = fromJsonString(parts[1], EnrollmentDTO.class);
+                Enrollment e = new Enrollment();
+                e.setEnrollmentId(dto.enrollmentId());
+                Section s = sectionRepository.findById(dto.sectionNo()).orElse(null);
+                User u = userRepository.findById(dto.studentId()).orElse(null);
+                if (s != null && u != null) {
+                    e.setSection(s);
+                    e.setStudent(u);
+                    e.setGrade(dto.grade());
+                    enrollmentRepository.save(e);
+                }
+
+            } else if (key.equals("deleteEnrollment")) {
+                enrollmentRepository.deleteById(Integer.parseInt(parts[1]));
+
+            } else {
+                System.out.println("Invalid action: " + key);
+            }
+
+        } catch (Exception e) {
+            System.out.println("Exception in receivedFromRegistrar " + e.getMessage());
+        }
+    }
+
+    private void sendMessage(String s) {
+        rabbitTemplate.convertAndSend(registrarServiceQueue.getName(), s);
+        System.out.println("Gradebook to Registrar " + s);
+    }
+    private static String asJsonString(final Object obj) {
+        try {
+            return new ObjectMapper().writeValueAsString(obj);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    private static <T> T  fromJsonString(String str, Class<T> valueType ) {
+        try {
+            return new ObjectMapper().readValue(str, valueType);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
Correctly sends the messages to the registrar. This method also converts the JSON to a string in order to send it as a message (String).